File: technical/reverse_c_style_string/golang/reverse_c_style_string.go
package reverse_c_style_string

func Reverse(b []byte) []byte {
	l := len(b)
	if l == 0 {
		return []byte{}
	}
	strlen := 0
	for int(b[strlen]) != 0 {
		strlen++
	}
	for i := 0; i < strlen/2; i++ {
		tmp := b[i]
		b[i] = b[strlen-i-1]
		b[strlen-i-1] = tmp
	}
	return b
}

File: technical/reverse_c_style_string/c++/reverse_c_style_string.cpp
#include <stdio.h>

void reverse(char* s) {
  int len = 0;
  while(s[len]) len++;
  if(len < 2) return;
  len--;
  int i = 0;
  while(i < len) {
    char tmp = s[i];
    s[i] = s[len];
    s[len] = tmp;
    i++;
    len--;
  }
}